trestlebot/tasks/sync_cac_content_task.py

```diff
@@ -91,6 +172,217 @@ def _add_props(self, oscal_component: DefinedComponent) -> DefinedComponent:
+                    continue
+
+                section_content_str = "\n".join(section_content)
+                section_content_str = pattern.sub("", section_content_str)
+                statement = self._create_statement(
+                    statement_id, section_content_str.strip()
+                )
+                implemented_req.statements.append(statement)
+        # else:
+        #     self._add_response_by_status(
+        #         implemented_req, oscal_status, control_response.strip()
+        #     )
+
+    def _process_rule_ids(self, rule_ids: List[str]) -> List[str]:
+        """
+        Process rule ids.
+        Notes: Rule ids with an "=" are parameters and should not be included
+        # when searching for rules.
+        """
+        processed_rule_ids: List[str] = list()
+        for rule_id in rule_ids:
+            parts = rule_id.split("=")
+            if len(parts) == 1:
+                processed_rule_ids.append(rule_id)
+        return processed_rule_ids
+
+    def _attach_rules(
+        self,
+        type_with_props: TypeWithProps,
+        rule_ids: List[str],
+    ) -> None:
+        """Add rules to a type with props."""
+        all_props: List[Property] = as_list(type_with_props.props)
+        # Get a subset from self.rules_by_id according to rule_ids
+        rules_by_id = {k: v for k, v in self.rules_by_id.items() if k in rule_ids}
+        rules: List[RuleInfo] = list(rules_by_id.values())
+        rules_transformer = RulesTransformer(
+            self.cac_content_root,
+            self.product,
+            self.cac_profile,
+        )
+        rule_properties: List[Property] = rules_transformer.transform(rules)
+        all_props.extend(rule_properties)
+        type_with_props.props = none_if_empty(all_props)
+
+    def _add_set_parameters(
+        self, control_implementation: ControlImplementation
+    ) -> None:
+        """Add set parameters to a control implementation."""
+        rules: List[RuleInfo] = list(self.rules_by_id.values())
+        params = []
+        for rule in rules:
+            params.extend(rule._parameters)
+        param_selections = {param.id: param.selected_value for param in params}
+
+        if param_selections:
+            all_set_params: List[SetParameter] = as_list(
+                control_implementation.set_parameters
+            )
+            for param_id, value in param_selections.items():
+                set_param = generate_sample_model(SetParameter)
+                set_param.param_id = param_id
+                set_param.values = [value]
+                all_set_params.append(set_param)
+            control_implementation.set_parameters = none_if_empty(all_set_params)
+
+    def _create_implemented_requirement(
+        self, control: Control
+    ) -> Optional[ImplementedRequirement]:
+        """Create implemented requirement from a control object"""
+
+        logger.info(f"Creating implemented requirement for {control.id}")
```

Comment:
How about move it to the line 351(after the `if control_id` )?

Reply:
It should be ok to leave it here imho. With this message, users could know which control is being processed, and if its validation failed, there will be another debug message following this one.